aten/src/ATen/ZeroTensorFallback.cpp

```diff
@@ -0,0 +1,99 @@
+          TORCH_CHECK(*is_write == alias_info->isWrite(),
+            "Unsupported operator for ", "ZeroTensorFallback: ", op.schema().name(),
+            "ZeroTensor fallback doesn't work for operators with a mix "
+            "mutable and non-mutable inputs that alias with outputs, "
+            "this must be implemented manually.  "
+            "If you got this error on a core op, please report a bug to PyTorch.");
+        } else {
+          is_write = alias_info->isWrite();
+        }
+      }
+    }
+
+    if (is_write.has_value() && !*is_write) {
+      op.redispatchBoxed(dispatch_keys & c10::DispatchKeySet(DispatchKeySet::FULL_AFTER, DispatchKey::ZeroTensor), stack);
+      return;
+    }
+
+    for (const auto i : c10::irange(num_arguments)) {
+      auto& ivalue = (*stack)[stack_start + i];
+      if (!(ivalue.isTensor() || ivalue.isTensorList())) {
+        continue;
+      }
+      const auto& argument = arguments[i];
+      bool mut_arg = false;
+
+      if (argument.alias_info()) {
+        // Was already tested by is_write loop above
+        TORCH_INTERNAL_ASSERT_DEBUG_ONLY(argument.alias_info()->isWrite());
+        mut_arg = true;
+      }
+
+      if (ivalue.isTensor()) {
+        auto tensor = std::move(ivalue).toTensor();
+        if (tensor.is_zerotensor()) {
+          TORCH_CHECK(!mut_arg, "ZeroTensors are immutable. Please use the materialized zero tensor ",
+                    "obtained using .clone() if you want a mutable tensor.");
+          tensor = tensor.clone();
+        }
+        (*stack)[stack_start + i] = std::move(tensor);
+      } else if (ivalue.isTensorList()) {
+        auto tensors = std::move(ivalue).toTensorList();
+        for(const auto j : c10::irange(tensors.size())) {
+          const Tensor& tensor = tensors[j];
+          if (tensor.is_zerotensor()) {
+            // TODO: assert requires_grad=False
+            //_like should not propagate zerotensor dispatch key
+            TORCH_CHECK(!mut_arg, "ZeroTensors are immutable. Please use the materialized zero tensor ",
+                    "obtained using .clone() if you want a mutable tensor.");
+            tensors[j] = tensor.clone();
+          }
+        }
+        (*stack)[stack_start + i] = std::move(tensors);
+      }
+    }
+
+    op.redispatchBoxed(dispatch_keys & c10::DispatchKeySet(DispatchKeySet::FULL_AFTER, DispatchKey::ZeroTensor), stack);
+  }
+
+
+  TORCH_LIBRARY_IMPL(_, ZeroTensor, m) {
+    m.fallback(torch::CppFunction::makeFromBoxedFunction<&zeroTensorFallback>());
+  }
+
+  TORCH_LIBRARY_IMPL(aten, ZeroTensor, m) {
+    m.impl("zeros_like", torch::CppFunction::makeFallthrough());
+    m.impl("empty_like", torch::CppFunction::makeFallthrough());
+    m.impl("empty_strided", torch::CppFunction::makeFallthrough());
+    m.impl("mul.Scalar", torch::CppFunction::makeFallthrough());
+    // specific registeration in native_functions.yaml
+    // m.impl("mul.Tensor", torch::CppFunction::makeFallthrough());
```

Comment:
Should this be removed?
If you want to document that some of these are specified in native functions, maybe just use a list? Or tell the reader to check that other file (so that the list doesn't go stale)?

I feel like having some as fallthrough and some as special definition in native_functions is a bit confusing. Should we just use one here?
If you prefer to keep both, it would be nice to have a comment here saying that we're doing both and the reader should not be surprised.